frontend/app/.server/domain/repositories/document-upload.repository.ts

```diff
@@ -25,21 +29,132 @@ export interface DocumentUploadRepository {
    * @returns A Promise that resolves to the document scan response entity.
    */
   scanDocument(documentScanRequestDto: DocumentScanRequestDto): Promise<DocumentScanResponseEntity>;
```

Comment:
Repository shouldn't reference service DTO even though they are the same structure.

```suggestion
  scanDocument(scanDocumentRequestEntity: ScanDocumentRequestEntity): Promise<ScanDocumentResponseEntity>;
```